products/BellHybrid/services/evtmgr/battery/BatteryController.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <evtmgr/battery/BatteryController.hpp>
+
+#include <module-utils/EventStore/EventStore.hpp>
+#include <service-evtmgr/BatteryMessages.hpp>
+#include <system/Constants.hpp>
+
+namespace sevm::battery::bell
+{
+    namespace
+    {
+        constexpr int lowVoltageThreshold{3400};
+    }
+
+    BatteryController::BatteryController(sys::Service *service,
+                                         xQueueHandle notificationChannel,
+                                         BatteryState::Thresholds thresholds)
+        : battery::BatteryController(service, notificationChannel, thresholds)
+    {}
+
+    void BatteryController::update()
+    {
+        battery::BatteryController::update();
+
+        const auto voltage = Store::Battery::get().voltage;
+        if (voltage < lowVoltageThreshold) {
+            sendLowVoltageShutdownMessage();
+        }
+    }
+
+    void BatteryController::sendLowVoltageShutdownMessage()
+    {
+        auto message = std::make_shared<sevm::BatteryLowVoltageShutdownMessage>();
+        getService()->bus.sendUnicast(std::move(message), ::service::name::system_manager);
+    }
+
+} // namespace sevm::battery::bell
```

Comment:
How about extending `BatteryState` class instead? I think it would be a good idea as it already contains battery state logic. Apart from that, PurePhone could benefit from this features too ;)

Reply:
Thanks for the advice. :) This approach was the fastest because we felt pressure due to the testing schedule.
I agree that should be moved to `BatteryState` so I'll try to rewrite this feature :)

Reply:
I checked the code and it looks like the `BatteryState` refer to predictable states. Brownout/low voltage is an unpredictable state so maybe it should be somewhere else? The Pure has its own class for brownout `BatteryBrownoutDetector`. This class is fed by `Events::Brownout` when the battery charger sends an interrupt.
What about extending this class and making an interface for Pure and Harmony for detecting brownout/low voltage?

Reply:
Done. I removed `BatteryController` for the bell and modified `BatteryBrownoutDetector`.